File: brand/screenshots/ios-app-guide/upload_to_drive.py
#!/usr/bin/env python3
"""Upload HTML files to Google Drive and make them publicly viewable."""

import os
import json
from google.oauth2 import service_account
from googleapiclient.discovery import build
from googleapiclient.http import MediaInMemoryUpload

# Service account key
SERVICE_ACCOUNT_FILE = os.path.expanduser("~/.config/google/pawbot-service-account.json")
SCOPES = ['https://www.googleapis.com/auth/drive.file']

# Use "PotLuck App Store Guide" folder
FOLDER_ID = "1occTURCFkbCpF30vXuVP4gKQ3qKC5xLP"

# Source files
SOURCE_DIR = "/home/orin_nano/.openclaw/workspace/potluck/brand/screenshots/ios-app-guide"

def get_credentials():
    """Load service account credentials."""
    credentials = service_account.Credentials.from_service_account_file(
        SERVICE_ACCOUNT_FILE, scopes=SCOPES)
    return credentials

def create_folder(service, name, parent_id):
    """Create a folder in Google Drive."""
    file_metadata = {
        'name': name,
        'mimeType': 'application/vnd.google-apps.folder',
        'parents': [parent_id]
    }
    folder = service.files().create(body=file_metadata, fields='id').execute()
    return folder['id']

def upload_file(service, file_path, folder_id):
    """Upload a file to Google Drive and make it publicly viewable."""
    filename = os.path.basename(file_path)
    
    with open(file_path, 'rb') as f:
        file_content = f.read()
    
    media = MediaInMemoryUpload(file_content, mimetype='text/html')
    
    file_metadata = {
        'name': filename,
        'parents': [folder_id],
        'mimeType': 'text/html'
    }
    
    file = service.files().create(
        body=file_metadata,
        media_body=media,
        fields='id, webViewLink, webContentLink'
    ).execute()
    
    # Make the file publicly viewable
    service.permissions().create(
        fileId=file['id'],
        body={
            'type': 'anyone',
            'role': 'reader'
        }
    ).execute()
    
    return file

def main():
    # Authenticate
    credentials = get_credentials()
    service = build('drive', 'v3', credentials=credentials)
    
    # Use the parent folder
    folder_id = FOLDER_ID
    print(f"Using folder ID: {folder_id}")
    
    # Upload files
    files = sorted([f for f in os.listdir(SOURCE_DIR) if f.endswith('.html')])
    uploaded_urls = []
    
    for filename in files:
        file_path = os.path.join(SOURCE_DIR, filename)
        print(f"Uploading: {filename}...")
        
        file = upload_file(service, file_path, folder_id)
        public_url = file.get('webViewLink', '')
        uploaded_urls.append((filename, public_url))
        print(f"  -> Uploaded! URL: {public_url}")
    
    print("\n" + "="*60)
    print("UPLOAD COMPLETE!")
    print("="*60)
    
    # Save URLs to a file for reference
    output_file = os.path.join(SOURCE_DIR, "uploaded_urls.txt")
    with open(output_file, 'w') as f:
        f.write("Uploaded Files:\n")
        f.write("="*60 + "\n")
        for filename, url in uploaded_urls:
            f.write(f"{filename}: {url}\n")
    
    print(f"\nURLs saved to: {output_file}")
    
    # Print summary
    print("\nUploaded files:")
    for filename, url in uploaded_urls:
        print(f"  {filename}")
        print(f"    {url}")

if __name__ == "__main__":
    main()

File: brand/screenshots/upload-to-drive.js
const { google } = require('googleapis');
const fs = require('fs');
const path = require('path');

const SCREENSHOT_DIR = '/home/orin_nano/.openclaw/workspace/potluck/brand/screenshots/appstore-guide';
const PARENT_FOLDER_ID = '1rMyOxeab6FBUDnuW-X2IV_dGLcEnmRJD';

const auth = new google.auth.GoogleAuth({
  keyFile: '/home/orin_nano/.config/google/pawbot-service-account.json',
  scopes: ['https://www.googleapis.com/auth/drive.file'],
});

const drive = google.drive({ version: 'v3', auth });

async function uploadAndShare(filePath, filename) {
  try {
    const fileMetadata = {
      name: filename,
      parents: [PARENT_FOLDER_ID],
    };

    const media = {
      mimeType: 'image/png',
      body: fs.createReadStream(filePath),
    };

    const file = await drive.files.create({
      resource: fileMetadata,
      media: media,
      fields: 'id',
      supportsAllDrives: true,
    });

    const fileId = file.data.id;

    await drive.permissions.create({
      fileId: fileId,
      requestBody: {
        type: 'anyone',
        role: 'reader',
      },
      supportsAllDrives: true,
    });

    const directLink = `https://drive.google.com/uc?id=${fileId}`;
    console.log(`Uploaded: ${filename} -> ${directLink}`);
    
    return { filename, fileId, directLink };
  } catch (error) {
    console.error(`Error uploading ${filename}:`, error.message);
    return null;
  }
}

async function main() {
  const files = fs.readdirSync(SCREENSHOT_DIR).filter(f => f.endsWith('.png'));
  
  const results = [];
  for (const file of files) {
    const filePath = path.join(SCREENSHOT_DIR, file);
    const result = await uploadAndShare(filePath, file);
    if (result) {
      results.push(result);
    }
  }
  
  console.log('\n--- Results ---');
  console.log(JSON.stringify(results, null, 2));
}

main().catch(console.error);
